src/solver/tron.jl

```diff
@@ -0,0 +1,403 @@
+#  Some parts of this code were adapted from
+# https://github.com/PythonOptimizers/NLP.py/blob/develop/nlp/optimize/tron.py
+
+using LinearOperators, NLPModels
+
+export tron, project!, project_step!, active, breakpoints
```

Comment:
I think `breakpoints`, `compute_Hs_slope_qs!`, `active`, `project!` and `project_step!` should be exported and moved to another file (`tools.jl`, `auxiliary.jl`, `utils.jl`?).
The functions `projected_line_search!` and `cauchy` are more specific to TRON, I think.
`projected_newton!` seems more general. Maybe another folder?